projects/arlas-toolkit/src/lib/components/histogram-tooltip-overlay/histogram-tooltip-overlay.component.html

```diff
@@ -1,40 +1,51 @@
-    <div class="interval">
-      <div class="axis">{{'interval:' | translate}}</div>
-      <div>{{tooltip.xRange.value}} {{(tooltip.dataType === 'time' ? tooltip?.xRange?.unit : tooltip.xUnit) | translate}}</div>
-    </div>
+<div class="tooltip">
+  <div class="tooltip__header header"><div class="header-title">{{tooltip.title | translate}}</div></div>
+   <div class="tooltip__xLabel label">
+     {{tooltip.xLabel | translate}} <span class="axis-label">({{'x-axis' | translate}})</span>
+   </div>
+   <div class="tooltip__xValue value-label">
+     <ng-container  *ngTemplateOutlet="intervalTemplate; context: {value: interval.start, isTime: tooltip.dataType === 'time' }"></ng-container>
+     <span class="interval-text" *ngIf="interval.end">to</span>
+     <ng-container *ngTemplateOutlet="intervalTemplate; context: {value: interval.end, isTime: tooltip.dataType === 'time' }"></ng-container>
+     <ng-container *ngTemplateOutlet="intervalTextTemplate;
+      context: {unit:tooltip.xUnit, interval:tooltip.xRange.unit, value: tooltip.xRange.value, isTime: tooltip.dataType === 'time'  }"></ng-container>
   </div>
-  <div class="xline">
-    <div class="labelwrapper">
-      <div class="axis">{{'x-axis:' | translate}}</div>
-    </div>
-    <div style="display: flex;">
-      <div class="xlabel">{{tooltip.xLabel | translate}}</div>
-      <div>{{(tooltip?.dataType === 'time' ? 'From' : '') | translate}} {{tooltip.xValue}} {{tooltip.xUnit}}</div>
-    </div>
+   <div class="tooltip__yLabel label">
+     {{tooltip.yLabel | translate}} <span class="axis-label">({{'y-axis' | translate}})</span>
   </div>
-
-  <div class="yline">
-    <div class="labelwrapper">
-      <div class="axis">{{'y-axis:' | translate}}</div>
-    </div>
-    <table class="values">
-      <tr *ngFor="let y of tooltip.y">
-        <td>
-          <div class="chip" [style.backgroundColor]="y.color"></div>
-        </td>
-        <td class="count">
-          <div>{{y.value}}</div>
-        </td>
-        <td  *ngIf="tooltip?.dataType === 'time'">
-          <div class="value_unit">{{y.chartId}}</div>
-        </td>
-        <td  *ngIf="tooltip?.dataType !== 'time' && tooltip.yUnit !== undefined && tooltip.yUnit !== ''">
-          <div class="value_unit">{{tooltip.yUnit}}</div>
-        </td>
-      </tr>
-    </table>
+   <div class="tooltip__yValue value-label">
+     <table class="values">
+       <tr *ngFor="let y of tooltip.y">
+         <td>
+           <div class="chip" [style.backgroundColor]="y.color"></div>
+         </td>
+         <td class="count">
+           <div>{{y.value}}</div>
+         </td>
+         <td  *ngIf="tooltip?.dataType === 'time'">
+           <div class="value-unit">{{(tooltip.yUnit && tooltip.yUnit .length > 0)?  tooltip.yUnit : y.chartId}}</div>
+         </td>
+         <td  *ngIf="tooltip?.dataType !== 'time' && tooltip.yUnit !== undefined && tooltip.yUnit !== ''">
+           <div class="value-unit">{{tooltip.yUnit}}</div>
```

Comment:
Shouldn't yUnit be translated ?

Reply:
same as Y i dont thinks. The value comme from an input text.

Reply:
They might be especially for the time histograms, where you can have for example the evolution over time of EO products which will be products or produits in French. For other types of histogram, I agree that it does not seem needed

Reply:
I sugggest to add a translate pipe too. This units come from configuration I agree but there are several times where we end up translatiing for a spacifi project even the input values.

Plus it will serve the day we will propose to fill translations in the builder